### PR TITLE
fix: Address issues with sequences, UN, and OW for WADO RetrieveMetadata

### PR DESCRIPTION
**Pull request by swederik**
This PR addresses issues with the current WADO-RS RetrieveMetadata logic as well as reorganizing the views so they are easier to debug.

# Changes:
- Moves the view function to a separate file and uses .toString() to pull them into CouchDB. It's much easier to debug the view functions if we keep them as separate files, we just have to remember that it uses something like ES5 because CouchDB uses SpiderMonkey by default.
- Fixes sequence support (this was breaking ImagePositionPatient, etc..)
- Fixes IntegerString and DecimalString values, which were previously returned as strings
- Adds InlineBinary for anything under 100 chars marked VR: UN
- Begins to add BulkDataURI support, but it has no root URL yet.
- Fixes broken patientBirthDate tag (was incorrectly using patientWeight, causing tests to fail)
- Fixes single value IntegerString and DecimalString entries by using dcmjs 0.8.3

I basically just pulled the same values from dcm4chee and did a diff to check what we are responding with vs. what it responds with:

```bash
curl https://server.dcmjs.org/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.2.840.113619.2.5.1762583153.215519.978957063.78/series/1.2.840.113619.2.5.1762583153.215519.978957063.121/instances/1.2.840.113619.2.5.1762583153.215519.978957063.128/metadata | python -m json.tool > dcm4chee.json

curl http://localhost:5985/studies/1.2.840.113619.2.5.1762583153.215519.978957063.78/series/1.2.840.113619.2.5.1762583153.215519.978957063.121/instances/1.2.840.113619.2.5.1762583153.215519.978957063.128/metadata | python -m json.tool > dicomweb-server.json

# Or use something visual to check the differences...
diff dicomweb-server.json dcm4chee.json
```

So far I've only tested with a single MR study, but I still think this is a step forward. We should add real tests for these functions to ensure we don't break them. It would also be good to do a smoke test to ensure that the views don't break when added to CouchDB, since improper JS (for SpiderMonkey) for the view functions will cause an immediate crash. You can't use e.g. arrow functions, object destructuring, etc... so we have to be careful with any refactoring for those.

These changes might help us get this PR into OHIF: https://github.com/OHIF/Viewers/pull/1122 so we can better control the data source we use for our end-to-end tests.

### CODE REVIEW

File: config/views.js [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
   wado_metadata: {
-    map:
-      "function(doc){var key={};if(doc.dataset){studyUID='NA';if (doc.dataset['0020000D'].Value[0]) studyUID=doc.dataset['0020000D'].Value[0];" +
-      " seriesUID='NA'; if (doc.dataset['0020000E'].Value[0]) seriesUID=doc.dataset['0020000E'].Value[0];" +
-      " instanceUID='NA'; if (doc.dataset['00080018'].Value[0]) instanceUID=doc.dataset['00080018'].Value[0];" +
-      " for(var t in doc.dataset){var fallback='';if(doc.dataset[t]&&doc.dataset[t].Value&&doc.dataset[t].Value[0]){key[t]={};if(doc.dataset[t].Value[0]!==''){if(doc.dataset[t].vr==='PN')" +
-      " key[t].Value=[{'Alphabetic':doc.dataset[t].Value[0]||fallback}];else key[t].Value=[doc.dataset[t].Value[0]||fallback]}" +
-      ' key[t].vr=doc.dataset[t].vr}}' +
-      ' emit([studyUID,seriesUID,instanceUID],key);}}',
+    map: wado.toString()
```
Comment (pieper) [inline config/views.js]: Agreed, much nicer

File: config/wado.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
+module.exports = function (doc) {
```
Comment (pieper) [inline config/wado.js]: Maybe add a comment here about the language restrictions and any other requirements?

Comment (emelalkim) [inline config/wado.js]: needs /* eslint-disable */ here

**Comment by dannyrb** [comment]
@emelalkim what commands do you run to see the test failure? Any qualms with CircleCI + GitHub status reporter to point out blocking failures?

File: config/wado.js [hunk 3]
```diff
@@ -83,6 +83,8 @@ module.exports = function (doc) {
         continue;
       }
 
+      console.log(Value);
```
Comment (dannyrb) [inline config/wado.js]: May want to nix this lingering `console.log`

**Comment by swederik** [comment]
Travis is acting weird, the CouchDB service stops responding after the STOW. Tests run locally so I'm going to merge this anyway. We should switch to CircleCI and use Docker.

Just use:
```bash
# Start the Database container
docker run -p 5984:5984 couchdb

# Run initial setup for CouchDB
curl -X PUT http://127.0.0.1:5984/_users
curl -X PUT http://127.0.0.1:5984/_replicator
curl -X PUT http://127.0.0.1:5984/_global_changes

# Run tests
yarn test
```

**Pull request merged**